src/Common/src/TypeSystem/Interop/IL/MarshalUtils.cs

```diff
@@ -15,7 +15,7 @@ public static bool IsBlittableType(TypeDesc type)
         {
             type = type.UnderlyingType;
 
-            if (type.IsValueType)
+            if (type.IsValueType && !type.IsNullable)
```

Comment:
This does not look right to me. Where is the equivalent condition in CoreCLR?

Reply:
You're right, this condition is a superfluous leftover from my experiments with debugging the exact shape of the conditions. Nullable types are automatically proclaimed as non-blittable because their "bool hasValue" field has the NFT_WINBOOL marshalling type and that ends up returning false from IsFieldBlittable at

https://github.com/dotnet/coreclr/blob/085398b5c6913b03180c855f56b967e2a7d6edde/src/vm/fieldmarshaler.cpp#L1126

I have reverted this unneeded condition in 2nd commit, thanks for pointing that out.